src/PKSim.Core/Services/CKDDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using OSPSuite.Core.Domain;
+using OSPSuite.Utility.Exceptions;
+using OSPSuite.Utility.Extensions;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+using static OSPSuite.Core.Domain.Constants.Parameters;
+using static PKSim.Core.CoreConstants.Organ;
+using static PKSim.Core.CoreConstants.Parameters;
+
+namespace PKSim.Core.Services
+{
+   public class CKDDiseaseStateImplementation : IDiseaseStateImplementation
+   {
+      private enum CKDStage
+      {
+         Stage3,
+         Stage4,
+         Stage5,
+      }
+
+      private readonly IValueOriginRepository _valueOriginRepository;
+      private readonly IDimensionRepository _dimensionRepository;
+      private const string TARGET_GFR = "TargetGFR";
+      private const int CKD_VALUE_ORIGIN_ID = 92;
+
+      public CKDDiseaseStateImplementation(
+         IValueOriginRepository valueOriginRepository,
+         IDimensionRepository dimensionRepository)
+      {
+         _valueOriginRepository = valueOriginRepository;
+         _dimensionRepository = dimensionRepository;
+      }
+
+      public bool IsSatisfiedBy(DiseaseState diseaseState) => diseaseState.IsNamed(CoreConstants.DiseaseStates.CKD);
+
+      public void ApplyTo(Individual individual)
+      {
+         var targetGFR = individual.OriginData.DiseaseStateParameters.FindByName(TARGET_GFR);
+
+         var organism = individual.Organism;
+         var kidney = organism.Organ(KIDNEY);
+         var fat = organism.Organ(FAT);
+         var smallIntestine = organism.Organ(SMALL_INTESTINE);
+         var bsa = organism.Parameter(BSA);
+         var hct = organism.Parameter(HCT);
+         var GFR_spec = kidney.Parameter(GFR_SPEC);
+         var kidneyVolume = kidney.Parameter(VOLUME);
+         var kidneySpecificBloodFlowRate = kidney.Parameter(SPECIFIC_BLOOD_FLOW_RATE);
+         var fatVolume = fat.Parameter(VOLUME);
+         var stomach = organism.EntityAt<IContainer>(LUMEN, STOMACH);
+         var plasmaProteinScaleFactorParameter = organism.Parameter(PLASMA_PROTEIN_SCALE_FACTOR);
+         var smallIntestinalTransitTime = smallIntestine.Parameter(SMALL_INTESTINAL_TRANSIT_TIME);
+         var gastricEmptyingTime = stomach.Parameter(GASTRIC_EMPTYING_TIME);
+
+         var bsaInm2 = bsa.ConvertToUnit(bsa.Value, CoreConstants.Units.m2);
+         var GFR_0 = GFR_spec.Value * kidneyVolume.Value * 1.73 / bsaInm2;
+         if (GFR_0 < targetGFR.Value)
+            throw new OSPSuiteException("ERROR");
```

Comment:
TODO: better error message

Reply:
I wonder what should we do if `GFR_0 = TargetGFR`? Should we continue or exit?
The most scaling factors would become 1 in this case, however some of them (e.g. hematocrit) might be <>1.
But in fact `GFR_0 = TargetGFR` means healthy individual, so I would not expect any further parameter changes in this case. Maybe smthg we should check with Paul.